shibboleth/middleware.py

```diff
@@ -56,8 +58,19 @@ def process_request(self, request):
             # User is valid.  Set request.user and persist user in the session
             # by logging the user in.
             request.user = user
+
+            # workaround for bug in Django < 1.10
+            # fixed in Django commit 3389c5ea2
+            if request.session.session_key is None:
+                django_version = django.utils.version.get_complete_version()
+                if django_version[0] == 1 and django_version[1] < 10:
+                    request.session.create()
+
             auth.login(request, user)
-            
+
+            # store session mapping
+            ShibSession.objects.get_or_create(shib=request.META['Shib_Session_ID'], session_id=request.session.session_key)
```

Comment:
@egroeper can you make this DB call only happen if the SLO stuff is enabled in app_settings?

Reply:
Done. It's really better to guard this as well.